### PR TITLE
DOC: update docstrings and links

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #553 and #631 (partial)

Fixes broken links in documentation and docstring formatting. 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?
```
sphinx-build -E -b html docs dist/docs
sphinx-build -b linkcheck docs dist/docs
```
Followed by visual inspection of local doc build.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Putting this up for discussion.

Open issues:
- [ ] `pysat.Instrument.bounds` and `pysat.Instrument.date` are defined "twice" due to property settings.  Feedback from sphinx suggests using `:noindex:` to resolve
- [ ] Example line in docstring code of 'inst-name-*-v??.cdf' triggers a warning, as rst thinks this is the start of an inline emphasis format statement.
- [ ] Switching CONTRIBUTING to an rst doc breaks the link check, as the rst does not yet exist in the main branch. 

The switch to rst for the contributing doc results from the line
```
* Preferably break long lines on open parentheses rather than using `\`

```
which rst interprets as ``.  Using `\\` allows this to display as expected.

**Comment by aburrell** [comment]
Switching Contributing to rst will likely also break the GitHub community profile.  I recommend not doing it.

**Comment by aburrell** [comment]
> Example line in docstring code of 'inst-name-*-v??.cdf' triggers a warning, as rst thinks this is the start of an inline emphasis format statement.
I believe you can get around this by putting the name in between grave accents.

**Comment by jklenzing** [comment]
> Switching Contributing to rst will likely also break the GitHub community profile. I recommend not doing it.

What if I rewrote the offending line using text?
```
* Preferably break long lines on open parentheses rather than using backslashes
```

**Comment by aburrell** [comment]
>> Switching Contributing to rst will likely also break the GitHub community profile. I recommend not doing it.
> What if I rewrote the offending line using text?

That's not the problem.  The community documents page only looks for markdown files.

**Comment by jklenzing** [comment]
Added issue #651 to document the :noindex: situation.  Assuming this passes, it's ready for review.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,19 +42,20 @@ To set up `pysat` for local development:
...
 
-   Tests for new instruments are performed automatically.  Tests for custom 
+   Tests for new instruments are performed automatically.  Tests for custom
    functions should be added to the appropriately named file in ``pysat/tests``.
-   For example, custom functions for the OMNI HRO data are tested in 
-   ``pysat/tests/test_omni_hro.py``.  If no test file exists, then you should 
-   create one.  This testing uses pytest, which will run tests on any python 
-   file in the test directory that starts with ``test``.  Classes must begin 
+   For example, custom functions for the OMNI HRO data are tested in
+   ``pysat/tests/test_omni_hro.py``.  If no test file exists, then you should
```
Comment (aburrell) [inline CONTRIBUTING.md:56]:
```suggestion
   For example, custom functions for the time utilities are tested in
   ``pysat/tests/test_utils_time.py``.  If no test file exists, then you should
```

**Review by aburrell — COMMENTED** [review]
Small change, since OMNI HRO is not a good example anymore!

**Review by aburrell — DISMISSED** [review]

**Comment by aburrell** [comment]
Actually, I found a link to fix from the Travis CI feedback:

(line    9) redirect  https://www.anaconda.com/distribution/ - permanently to https://www.anaconda.com/products/individual

**Comment by jklenzing** [comment]
Fixed.

**Comment by jklenzing** [comment]
Also caught an http to https for icon

**Comment by jklenzing** [comment]
Ready for review

**Review by aburrell — APPROVED** [review]

**Pull request merged**